ExceptionsHomework/src/ro/scit/menu/Menu.java

```diff
@@ -0,0 +1,160 @@
+                            student.setGender(GenderEnum.FEMALE.getGender());
+                        } else {
+                            System.out.println("Invalid option! Please try again.");
+                        }
+                        Student student1 = new Student(cnp,firstName, lastName, birthDate, age, gender);
+                        addStudent(student1);
+                        System.out.println("Your Student has been added");
+                    }catch (NumberFormatException e){
+                        System.err.println("Please insert a number");
+                    }
+
+//                    try {
+//                        Date date = sdf.parse(str);
+//                        sdf = new SimpleDateFormat("dd.MM.yyyy");
+//                        System.out.println("Date of birth: " + sdf.format(date));
+//                    } catch (ParseException e) {
+//                        System.out.println("Parse Exception");
+//                    }
+
+
+                    break;
+                case 2:
+                    System.out.println("Insert the student's cnp");
+                    String cnp1 = sc.nextLine();
+                    deleteStudent(cnp1);
+                    System.out.println("The student has been deleted! ");
+
+                    break;
+                case 3:
+                    System.out.println("Insert the age: ");
+                    int age = Integer.parseInt(sc.nextLine());
+                    System.out.println("all students with age " + age + " : ");
+                    retrievebyAge(age);
+                    break;
+                case 4:
+                    System.out.println("List Sorted by name");
+                    StudentNameComparator nameCompare = new StudentNameComparator();
+                    Collections.sort(studentList, nameCompare);
+                    for (Student s: studentList)
+                        System.out.println(s);
+                    break;
+                case 5:
+                    this.exit();
+                    break;
+                default:
+                    System.out.println("Invalid selection.");
+                    break;
+
+            }
+        }
+    }
+
+    private void exit () {
+        System.out.println("Exiting...");
+        System.exit(1);
+
+    }
+
+    //Add students in list
+    public  boolean addStudent (Student student){
+        if( student.getFirstName().isEmpty() || student.getLastName().isEmpty()){
+            throw new IllegalArgumentException("The first name or last name should not be empty!");
+        }else {
+            studentList.add(student);
+            return true;
+        }
+    }
+    //Remove students from list
+    public  boolean deleteStudent (String cnp){
+        if( cnp.isEmpty()){
+            throw new IllegalArgumentException("The cnp should not be empty!");
+        }else {
+            Iterator<Student> iterator = studentList.iterator();
```

Comment:
It's great that you're using an `Iterator` here. You correctly recognized that we shouldn't remove elements from a collection while iterating over it using the traditional for-loop; because the operation might throw a `ConcurrentModificationException`.